flow/EncryptUtils.cpp

```diff
@@ -61,7 +61,9 @@ std::string getEncryptDbgTraceKeyWithTS(std::string_view prefix,
                                         int64_t expAfterTS) {
 	// Construct the TraceEvent field key ensuring its uniqueness and compliance to TraceEvent field validator and log
 	// parsing tools
+	std::string dName = domainName.toString();
+	// Underscores are invalid in trace event detail name.
+	boost::replace_all(dName, "_", "-");
```

Comment:
The only characters legal in a trace event detail are A-Za-z0-9. You'll need to exclude all punctuation from your detail name. My suggestion is to split this into two fields so that what is now your field key becomes a field value.

Reply:
Sounds good, if that is OK, I will make it as a separate change (as it impacts EKP logging more than this PR proposed changes).
Thanks for pointing out.